### DISCUSSION TITLE
Autoupgrade module downloads all native modules to disk

### DISCUSSION THREAD

**Discussion by Hlavtox** [PrestaShop/PrestaShop#34447] [Ideas]
### Describe the bug and add attachments

If you upgrade using autoupgrade module, it will download all native modules into your installation, bloating your modules folder.

I had 59 modules before upgrading.
I have 107 modules after upgrading.

### All of these were downloaded:
```
blockreassurance
blockwishlist
contactform
dashactivity
dashgoals
dashproducts
dashtrends
followup
graphnvd3
gridhtml
pagesnotfound
productcomments
ps_banner
ps_brandlist
ps_cashondelivery
ps_customtext
ps_dataprivacy
ps_distributionapiclient
ps_faviconnotificationbo
ps_featuredproducts
ps_checkpayment
ps_mainmenu
ps_reminder
ps_searchbar
ps_supplierlist
ps_themecusto
ps_wirepayment
psaddonsconnect
psgdpr
referralprogram
statsbestcategories
statsbestcustomers
statsbestmanufacturers
statsbestproducts
statsbestsuppliers
statsbestvouchers
statscarrier
statscatalog
statsdata
statsforecast
statscheckup
statsnewsletter
statspersonalinfos
statsproduct
statsregistrations
statssales
statssearch
statsstock
```

### Expected behavior

No extra module should be installed.

### Steps to reproduce

1. Install 1.7.6.9
2. Remove all stat modules, dash modules and some others.
3. Check your `/modules` folder that these modules are not there.
4. Upgrade to 8.1.2 and check that your `/modules` folder is bloated with uninstalled modules you don't need and didn't have before.

### PrestaShop version(s) where the bug happened

1.7.6.9 -> 8.1.2, but probably happens during all upgrades

### PHP version(s) where the bug happened

doesn't matter

**Comment by MatShir** [comment]
Make it configurable I guess, or only for new module. I think if we have new module native to be there after the upgrade, like for our new api module.

**Comment by Misty-94** [comment]
It's also a security problem because as only the files are present and the modules are not installed they will never be updated when there are new versions of these modules available. Also as these modules are not installed there is no way in the backoffice to delete these files, you have to delete them manually with each upgrade.
I upgraded from 1.7.8.7 to 1.7.8.10 and had the same problem : I unistalled a lot of modules in 1.7.8.7 (with option to delete files) and when I upgraded to 1.7.8.10 all native modules are now in the modules directory but not installed in the backoffice, so I had to delete them manually. So at the moment the option to uninstall with option to delete all files is dangerous for security.

**Comment by paulnoelcholot** [comment]
Hello @Hlavtox,

Thank you for your suggestion. The Maintainer Team will take it into consideration for future developments. 

Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [contacting a professional](https://www.prestashop-project.org/support/) to help you create it. 

If you or anyone else is willing to [contribute this feature](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/) to the PrestaShop Project, the teams will be happy to help by ensuring that it fits well within the project. Please let us know if that is the case!

Thank you

**Comment by Hlavtox** [comment]
@M0rgan01 @Quetzacoalt91 Guys, can you check what is the current state of this behavior? Is it removed in the latest versions of autoupgrade module after your refactor? :-)

**Comment by Hlavtox** [comment]
Ping @M0rgan01 @Quetzacoalt91 Guys, still the same behavior with https://github.com/PrestaShop/autoupgrade/releases/tag/v7.0.0-dev.4 and 9.0.0, do you think it could be improved?

Having uinstalled modules lying on the disk is a bit of a security risk, and just a bit of an annoyance to delete them after upgrade.
